src/main/scala/edu/ie3/simona/event/listener/ResultEventListener.scala

```diff
@@ -299,18 +303,20 @@ class ResultEventListener(
             resultFileHierarchy
           )
         )
-        .onComplete {
-          case Failure(exception) =>
-            throw new InitializationException(
-              "Cannot initialize result sinks!"
-            ).initCause(exception)
-            self ! PoisonPill
-          case Success(classToSink) =>
-            log.debug("Initialization complete!")
-            supervisor ! ServiceInitComplete
-            self ! BaseData(classToSink.toMap)
-        }
+        .map(result => SinkResponse(result.toMap))
+        .pipeTo(self)
       stay()
+
+    case Event(SinkResponse(classToSink), _) =>
+      // Sink Initialization succeeded
+      log.debug("Initialization complete!")
+      supervisor ! ServiceInitComplete
+
+      unstashAll()
+      goto(Idle) using BaseData(classToSink)
+
+    case Event(Status.Failure(ex), _) =>
+      throw new InitializationException("Unable to setup SimonaSim.", ex)
```

Comment:
Maybe an additional logging statement with the exception could be slightly more verbose here?

Reply:
Good point, please have a look again

Reply:
There also is a logging statement in the defined supervision strategy. We might have a look, if this is not double the work, then.

Reply:
seems it would log twice

```
      case ex: Exception =>
        self ! EmergencyShutdownInitiated
        logger.error(
          "The simulation's guardian received an uncaught exception. - {}: \"{}\" - Start emergency shutdown.",
          ex.getClass.getSimpleName,
          ex.getMessage
        )
```